### PR TITLE
[framework] oracle and fix bcs and refactor utxo event

### PR DESCRIPTION
**Pull request by mx819812523**
## Summary

1.  sync sui oracle
2. fix bcs
3. refactor bitcoin utxo event

- Closes #issue

### CODE REVIEW

File: frameworks/rooch-framework/sources/oracle/oracle_meta.move [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+            } else {
+                return 2
+            }
+        }else {
+            assert!(false, EUnsupportedDataType)
+        };
+        0
+    }
+
+    public fun quick_sort<T: drop + copy>(data: vector<T>): vector<T> {
```
Comment (jolestar) [inline frameworks/rooch-framework/sources/oracle/oracle_meta.move:144]: use `moveos_std::sort::sort`

**Comment by jolestar** [comment]
We need some tests for the bcs fix and Oracle.

File: frameworks/rooch-framework/sources/oracle/oracle_meta.move [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+    /// take the median value
+    public fun median<T: copy + drop>(meta_oracle: MetaOracle<T>): TrustedData<T> {
+        let (values, oracles) = combine(meta_oracle);
+        let sortedData = quick_sort(values);
+        let i = vector::length(&sortedData) / 2;
+        let value = vector::remove(&mut sortedData, i);
+        TrustedData { value, oracles }
+    }
+
+    fun cmp<T: copy + drop>(a: &T, b: &T): u8 {
```
[the remark this answers is not in the document]
Comment (mx819812523) [reply frameworks/rooch-framework/sources/oracle/oracle_meta.move:87]: need move DecimalValue to std

**Review by jolestar — APPROVED** [review]

File: frameworks/rooch-framework/sources/oracle/oracle.move [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+            timestamp: now_milliseconds(),
+            identifier,
+        };
+        object::add_field(&mut oracle.id, ticker, new_data);
+    }
+
+    public entry fun archive_data<K: store + copy + drop, V: store + copy + drop>(
+        oracle_obj: &mut Object<SimpleOracle>,
+        ticker: String,
+        archival_key: K,
```
Comment (jolestar) [inline frameworks/rooch-framework/sources/oracle/oracle.move:104]: K also can not be tx argument.

**Review by jolestar — APPROVED** [review]

**Pull request merged**